Classes/Domain/Model/Dto/RecordInstanceIdentifier.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pixelant\Interest\Domain\Model\Dto;
+
+use Pixelant\Interest\Domain\Model\Dto\Exception\InvalidArgumentException;
+use Pixelant\Interest\Utility\TcaUtility;
+use TYPO3\CMS\Core\Site\Entity\SiteLanguage;
+use TYPO3\CMS\Core\Site\SiteFinder;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+/**
+ * DTO to handle record instance identifier.
+ */
+class RecordInstanceIdentifier
+{
+    public const LANGUAGE_ASPECT_PREFIX = '|||L';
+
+    /**
+     * @var string
+     */
+    protected string $table;
+
+    /**
+     * The original remote id from construct.
+     *
+     * @var string
+     */
+    protected string $remoteId;
+
+    /**
+     * Language to use for processing.
+     *
+     * @var SiteLanguage|null
+     */
+    protected ?SiteLanguage $language;
+
+    /**
+     * @var string|null
+     */
+    protected ?string $workspace;
+
+    /**
+     * @param string $table
+     * @param string $remoteId
+     * @param string|null $language as RFC 1766/3066 string, e.g. nb or sv-SE.
+     * @param string|null $workspace workspace represented with a remote ID.
+     */
+    public function __construct(
+        string $table,
+        string $remoteId,
+        ?string $language = null,
+        ?string $workspace = null
+    ) {
+        $this->table = strtolower($table);
+        $this->remoteId = $remoteId;
+        $this->workspace = $workspace;
+        $this->language = $this->resolveLanguage((string)$language);
```

Comment:
Is there a need to typecast this as it already is a string?

Reply:
It's a string *or* null, so by casting null to string we get an empty string. The question would be if this conversion should happen earlier.

Reply:
I have changed this now.